### PR TITLE
Feature - Code clean up and Unit testing for results screen

### PR DESCRIPTION
**Pull request by ksookdhew**
## Description 
Code clean up and Unit testing for results screen

## Checklist 
Please review and check the following before submitting this pull request: 
- [x] The code follows the project's coding standards. 
- [x] 100% Unit test coverage for View Models 
- [x] All warnings Resolved.

Note: Will change the names in the model at a later stage

### CODE REVIEW

File: F1/F1/Results/RacingResultViewController.swift [hunk 1]
```diff
@@ -8,40 +8,46 @@ import UIKit
...
 
+    // MARK: Variables
+    var race: Race?
+
+    // MARK: Functions
     override func viewDidLoad() {
         super.viewDidLoad()
         setupTableView()
-        viewModel.setRaceResult(raceRes: race!)
+        viewModel.setRaceResult(raceRes: race)
```
Comment (giacatano) [inline F1/F1/Results/RacingResultViewController.swift:24]: Not sure what raceRes is. Maybe check the naming convention.

**Review by giacatano — APPROVED** [review]

File: F1/F1/Results/RacingResultViewController.swift [hunk 2]
```diff
@@ -8,40 +8,46 @@ import UIKit
 
 class RacingResultViewController: UIViewController {
 
-    var round = ""
-    var race: Race?
+    private lazy var viewModel = ResultsViewModel(repository: ResultsRepository(), delegate: self)
+
+    // MARK: IBOutlets
     @IBOutlet weak var raceName: UILabel!
```
Comment (kieran-woodrow) [inline F1/F1/Results/RacingResultViewController.swift:14]: private :)

File: F1/F1/Results/ResultsViewModel.swift [hunk 3]
```diff
@@ -19,12 +21,23 @@ class ResultsViewModel {
...
         return allResults?.count ?? 0
     }
 
+    var raceResultsCount: Int {
+        raceResult?.count ?? 0
+    }
+
+    var raceName: String {
+        race?.raceName ?? "Race Name"
+
```
Comment (kieran-woodrow) [inline F1/F1/Results/ResultsViewModel.swift:35]: extra new line can be removed

File: F1/F1/Results/ResultsViewModel.swift [hunk 4]
```diff
@@ -19,12 +21,23 @@ class ResultsViewModel {
...
+    }
+
+    // MARK: Functions
     func allResult(atIndex: Int) -> Race? {
-        return allResults?[atIndex] ?? nil
+        allResults?[atIndex] ?? nil
```
Comment (kieran-woodrow) [inline F1/F1/Results/ResultsViewModel.swift:40]: just checking, you do you want to return nil if there is nothing there?

File: F1/F1/Results/ResultsViewModel.swift [hunk 5]
```diff
@@ -68,19 +72,19 @@ class ResultsViewModel {
...
         }
-
     }
 
     func fetchResults() {
-        repository?.fetchRacingResults(completion: { [weak self] result in
+        repository?.fetchRacingResults { [weak self] result in
             switch result {
             case .success(let result):
+                // TODO: Change mrData
```
Comment (kieran-woodrow) [inline F1/F1/Results/ResultsViewModel.swift:81]: still the todo?

Comment (kieran-woodrow) [reply F1/F1/Results/ResultsViewModel.swift:81]: nice use of trailing closure as well! :)

File: F1/F1Tests/ResultsViewModelTests.swift [hunk 6]
```diff
@@ -0,0 +1,155 @@
+//
+//  ResultsViewModelTests.swift
+//  F1Tests
+//
+//  Created by Kaitlyn Sookdhew on 2024/05/04.
+//
+class MockResultsRepository: ResultsRepositoryType {
```
Comment (kieran-woodrow) [inline F1/F1Tests/ResultsViewModelTests.swift:7]: nice!

**Review by kieran-woodrow — CHANGES_REQUESTED** [review]

File: F1/F1/Results/AllResultsTableViewCell.swift [hunk 7]
```diff
@@ -9,45 +9,48 @@ import UIKit
...
-        self.driver1.layer.masksToBounds = true
-        self.driver2.layer.cornerRadius = 8
-        self.driver2.layer.masksToBounds = true
-        self.driver3.layer.cornerRadius = 8
-        self.driver3.layer.masksToBounds = true
+        setUpCell()
+    }
 
+    func setUpCell() {
+        contentView.layer.cornerRadius = 10
```
[the remark this answers is not in the document]
Comment (ksookdhew) [reply F1/F1/Results/AllResultsTableViewCell.swift:29]: Thanks will definitely have a look

File: F1/F1/Results/RaceResultTableViewCell.swift [hunk 8]
```diff
@@ -9,12 +9,11 @@ import UIKit
...
-    @IBOutlet weak var points: UILabel!
+    @IBOutlet weak private var position: UILabel!
+    @IBOutlet weak private var driver: UILabel!
+    @IBOutlet weak private var time: UILabel!
+    @IBOutlet weak private var points: UILabel!
 
-    static let identifier = "RaceResultTableViewCell"
     override func awakeFromNib() {
         super.awakeFromNib()
         self.time.layer.cornerRadius = 8
```
[the remark this answers is not in the document]
Comment (ksookdhew) [reply F1/F1/Results/RaceResultTableViewCell.swift:19]: Definitely will get back to this. I know in my later code I use the storyboard more, its just this is my early code :)

File: F1/F1/Results/ResultsViewModel.swift (hunk 5 above)
Comment (ksookdhew) [reply F1/F1/Results/ResultsViewModel.swift:80]: Changing the name in this PR meant adding a lot more files in one PR. Going to change it in the next PR

File: F1/F1/Results/ResultsViewModel.swift (hunk 4 above)
Comment (ksookdhew) [reply F1/F1/Results/ResultsViewModel.swift:39]: Yes I do 🙈

**Review by kieran-woodrow — APPROVED** [review]

**Review by sjacobs1 — APPROVED** [review]

**Pull request merged**